R/private-lib.R

```diff
@@ -121,7 +143,8 @@ lookup_deps <- function(package, lib_path = .libPaths()) {
   }
 
   while (length(todo)) {
-    new <- unlist(lapply(todo, extract_deps))
+    new <- unlist(lapply(todo, extract_deps, soft = soft))
+    soft <- FALSE
```

Comment:
Just unconditionally setting this to FALSE after the first time seemed the easiest change without refactoring how the loop.